src/js/relayclient/ServerHelper.js

```diff
@@ -1,29 +1,40 @@
 const BN = require('web3').utils.toBN
 
+const { event2topic } = require('./utils')
 // relays are "down-scored" in case they timed out a request.
 // they are "forgiven" after this timeout.
 const DEFAULT_RELAY_TIMEOUT_GRACE_SEC = 60 * 30
 
+// assume veri high stake values for preferred relays, so we don't need
+// to change the filter logic to always support them.
+const PREFERRED_STAKE = 1e30
+const PREFERRED_DELAY = 1e10
 class ActiveRelayPinger {
   // TODO: 'httpSend' should be on a network layer
-  constructor (filteredRelays, httpSend, gasPrice, verbose) {
+  constructor (filteredRelays, httpSend, gasPrice, verbose, nextPinger) {
     this.remainingRelays = filteredRelays.slice()
     this.httpSend = httpSend
     this.pingedRelays = 0
     this.relaysCount = filteredRelays.length
     this.gasPrice = gasPrice
     this.verbose = verbose
+    this.nextPinger = nextPinger
   }
 
   /**
    * Ping those relays that were not returned yet. Remove the returned relay (first to respond) from {@link remainingRelays}
    * @returns the first relay to respond to a ping message. Note: will never return the same relay twice.
    */
   async nextRelay () {
-    if (this.remainingRelays.length === 0) {
-      return null
+    let ret = await this.nextRelay1()
+    // only if no relay found in this ActivePinger, start using the next ActivePinger
+    if (!ret && this.nextPinger) {
+      ret = await this.nextPinger.nextRelay()
```

Comment:
Weird

Reply:
I wanted the pinger mechanism to run first on the list of "preferredRelays", and only if none is found, start with the normal relays.
At first, I simply put the preferred relays at the beginning of the list. But then it occurred to me that if there's exactly 1 (or two) preferred ones, and we ping 3 simultaneously, you might start using "non-preferred" relay just because it answered the ping earlier.
So instead, I added this chaining of pingers.

Reply:
In this case, I would consider a different approach
Existing `ActiveRelayPinger` tries the first 3 relays because it is told to do so. If it could differentiate between 'preferred' and 'fallback' relays, it would be able to contain the etire business logic of it's class, instead of spreading it between two nested instances of itself.
Consider smth like:
```    let slice = this.preferredRelays.slice(0, bulkSize)
    if (slice.length === 0){
       slice = this.fallbackRelays.slice(0, bulkSize)
```

Reply:
that's not enough: after trying a relay, we need to remove it from the list, and try with the rest. this logic on handling "remainingRelays" need to be duplicated into "preferredRelays".

I think its more duplicated code than adding nextPinger